template/src/core/network/async-process/asyncProcess.d.ts

```diff
@@ -0,0 +1,26 @@
+interface AsyncProcessState<Data = any, Payload = any> {
+  isRequestPending: boolean;
+  isRequestFetched: boolean;
+  data: Data | null;
+  error:
+    | (Error & {
+        data: any;
+        status: number;
+      })
+    | null;

```

Comment:
It should be of type `Error | null`
```suggestion
  error: Error | null;
```

We should aim for consistent typing and an error should be an instanceof the Error object